### ISSUE TITLE
Feature: Optionale Möglichkeit zur Auswahl der Inhaltselemente per Icon

### ISSUE THREAD

**Issue by BennyK80** [contao/contao#721]
Hallo zusammen,

ein Feature, welches eine leichtere und schnellere Auswahlmöglichkeit eines Inhaltselementes ermöglichen würde, wäre eine optionale Darstellung mit Icons.

Einige meiner Kunden sind überfordert mit den Begriffen im Select-Menü und der Verwendeung des Select-Menüs selbst. Es ist nicht immer ganz klar, dass dort etwas ausgewählt werden kann.
(Für mich schon, für den Kunden nicht immer.)

Nach ein paar Wochen Contao-Pause, haben es einige wieder vergessen und benötigen Hilfe.

Worpress hat das meiner Meinung nach sehr schön gelöst.
https://de.wordpress.org/gutenberg/

Daher die Überlegung, ob so etwas nicht auch in Contao nützlich wäre.
z.B. so oder ähnlich.

![Contao_Inhaltselemente-Auswahl](https://user-images.githubusercontent.com/7669964/64268359-37a21c80-cf38-11e9-88ae-5ea7f20b3515.png)

Wenn kein Icon hinterlegt wurde (z.B. bei Custom Elemnts) wird ein Platzhalter angezeigt.

Natürlich sehe ich auch den paltzsparenden Vorteil der aktuellen Lösung.
Das Contao kein Homepage-Baukasten ist, ist mir ebenfalls bewusst.
Vielleicht ist es auch nicht im Sinne des Erfinders.
Eventuell gab es auch schon Diskussionen dazu?

Daher nur optional!
z.B. in den Backend-Einstellungen des Benutzerprofils.
"Inhaltselemente mit Icons anzeigen"

Ich bin kein Entwickler und kann somit nicht den Arbeitsaufwand dahinter beurteilen.
Wie sehen das andere?

Viele Grüße,
Benny

**Comment by aschempp** [comment]
@qzminski might have a solution for this 🙃

**Comment by leofeyer** [comment]
I like the idea.

**Comment by asaage** [comment]
i like the grouped chosen select very much.
There are almost 30 element-types in a basic install - i'm afraid this will clutter up screen-space pretty quick. Might make sense with the "allowed content-types feature" however - but it should be optional.

**Comment by qzminski** [comment]
For one of our commercial projects that is yet to be released we (credits to @spin80 @milbau) introduced a similar solution.

Content element widget:
![Zrzut ekranu 2019-09-9 o 12 38 06](https://user-images.githubusercontent.com/193483/64524503-e2328a80-d2fe-11e9-98ec-384f7e557757.png)

Modal window:
![Zrzut ekranu 2019-09-9 o 12 39 29](https://user-images.githubusercontent.com/193483/64524504-e2cb2100-d2fe-11e9-9481-ddd6597a97ea.png)

**Comment by fritzmg** [comment]
@qzminski wow, that's pretty nice. And it doesn't even require more clicks than the chosen select, right? It's just _click » type » click_ for both.

**Comment by qzminski** [comment]
That's right @fritzmg 🙂Forgot to mention the text input at the top is a search bar and buttons below are responsible for filtering per category… but I think it's obvious 😅

**Comment by fritzmg** [comment]
👍 
Did you also extend the DCA for custom icons? Or is it just CSS via the type of the element.

**Comment by qzminski** [comment]
Because it's an integral part of our bigger extension, the icon files have to be named exactly after the content element types (e.g. `elements/text.svg` or so).

**Comment by BennyK80** [comment]
Ich habe gerade folgende Erweiterung entdeckt:

https://github.com/leycommediasolutions/contao-frontend-builder

Das Feature ist für mich daher nicht mehr notwendig.

**Comment by fritzmg** [comment]
Imho this issue should stay open.

**Comment by zonky2** [comment]
here a screenshot from T3

![](https://user-images.githubusercontent.com/1192057/132835921-cd07b65e-a3f3-4c5d-8199-0cd726738091.png)
